src/fastify-plugins/maps/static-maps.js

```diff
@@ -0,0 +1,266 @@
+              ).href,
+            }
+          })
+        )
+      ).filter(
+        /**
+         * @template {import('@sinclair/typebox').Static<typeof ListStaticMapsReplySchema>[number] | null} V
+         * @param {V} v
+         * @returns {v is NonNullable<V>}
+         */
+        (v) => v !== null
+      )
+
+      return result
+    }
+  )
+
+  /// Get a map's style.json
+  fastify.get(
+    `/:styleId/style.json`,
+    { schema: { params: GetStyleJsonParamsSchema } },
+    async (req, rep) => {
+      const { styleId } = req.params
+
+      /** @type {import('fs').Stats} */
+      let stat
+
+      /** @type {string | Buffer} */
+      let data
+
+      try {
+        const filePath = path.join(staticRootDir, styleId, 'style.json')
+        stat = await fs.stat(filePath)
+        data = await fs.readFile(filePath, 'utf-8')
+      } catch (err) {
+        throw new NotFoundError(`id = ${styleId}, style.json`)
+      }
+
+      const serverAddress = await getFastifyServerAddress(req.server.server)
+
+      data = Buffer.from(
+        data.replace(
+          /\{host\}/gm,
+          new URL(`${fastify.prefix}/${styleId}`, serverAddress).href
+        )
+      )
+      rep.header('Content-Type', 'application/json; charset=utf-8')
+      rep.header('Last-Modified', new Date(stat.mtime).toUTCString())
+      rep.header('Cache-Control', 'max-age=' + 5 * 60) // 5 minutes
+      rep.header('Content-Length', data.length)
+      rep.header(
+        'Access-Control-Allow-Headers',
+        'Authorization, Content-Type, If-Match, If-Modified-Since, If-None-Match, If-Unmodified-Since'
+      )
+      rep.header('Access-Control-Allow-Origin', '*')
+
+      rep.send(data)
+    }
+  )
+
+  // Get a tile (extension specified)
+  fastify.get(
+    `/:styleId/tiles/:tileId/:z/:x/:y.:ext`,
+    { schema: { params: GetStaticMapTileParamsSchema } },
+    handleStyleTileGet
+  )
+  // Get a tile (extension not specified)
+  fastify.get(
+    `/:styleId/tiles/:tileId/:z/:x/:y`,
+    { schema: { params: GetStaticMapTileParamsSchema } },
+    handleStyleTileGet
+  )
+}
+
+/**
+ * @param {string} archive
+ * @param {string} filename
+ */
+function extractAsarFile(archive, filename) {
+  try {
+    return asar.extractFile(archive, filename)
+  } catch (err) {
+    return undefined
+  }
+}
+
+const mime = new Mime(standardTypes, { 'application/x-protobuf': ['pbf'] })
+
+/**
+ * @param {string} baseDirectory
+ * @param {import('@sinclair/typebox').Static<typeof GetStaticMapTileParamsSchema>} params
+ * @returns {null | { data: Buffer, mimeType: string | null, shouldGzip: boolean}}
+ */
+function getStyleTileInfo(baseDirectory, params) {
+  const { styleId, tileId, z, x, y } = params
+  let { ext } = params
+
+  // TODO: If necessary, need to flip the y value first if the TileJSON source is TMS scheme
+  // Doing this will depend on if we decide that the asar directory structure should only follow XYZ or if it should align with corresponding tilejson spec
```

Comment:
highlighting this. are we able to assume that the tile urls use z/x/y format and the directories that are stored on the filesystem follow that convention?

Reply:
Not my most helpful comment, but want to explicitly say: I don't know. My hunch is that we should get an answer to this before the MVP, but not before merging this.